pyfr/backends/openmp/xsmm.py

```diff
@@ -77,33 +78,42 @@ def mul(self, a, b, out, alpha=1.0, beta=0.0):
                        out.nbytes >= 32*1024**2 and
                        self.backend.alignb >= 64)
 
-            a_np = np.ascontiguousarray(a.get())
-            m, k = a_np.shape
-
             if self.backend.fpdtype == np.float64:
                 xsmm_dtype = 0
                 alpha, beta = c_double(alpha), c_double(beta)
             else:
                 xsmm_dtype = 1
                 alpha, beta = c_float(alpha), c_float(beta)
 
-            # JIT and register an block leaddim size kernel for this matrix
-            blkptr = self._wrappers.libxsmm_fsspmdm_create(
-                xsmm_dtype, m, b.leaddim, k, k, ldb, ldc, byref(alpha),
-                byref(beta), c_is_nt, a_np.ctypes.data
-            )
+            blkptr = []
+            for fac in a_facs:
+                mat = np.ascontiguousarray(fac.get())
+                m, k = mat.shape
+
+                if fac == a_facs[0]:
```

Comment:
Enumerate and then do i == 0.  Can even do a pair of inline if's for assigning beta and c_is_nt.